include/cantera/base/fmt.h

```diff
@@ -1,16 +1,47 @@
 //! @file fmt.h Wrapper for either system-installed or local headers for fmt
+#ifndef CT_FMT_H
+#define CT_FMT_H
 #include "ct_defs.h"
 
+//! Do not use the fmt macro from fmtlib because it shadows a function of
+//! the same name in kinetics/Group.h
+#define FMT_NO_FMT_STRING_ALIAS
+
+//! Only use the header portions of fmtlib
+#define FMT_HEADER_ONLY
```

Comment:
Is there a reason we're changing to using this as header only? We were specifically compiling the compilable portion before.

Reply:
Well, the short answer is because I didn't know that and this fixed the compile problems I was having. I think I fixed this now, but fmtlib >= 5.0.0 is apparently distributed with a pre-compiled static library (at least from Homebrew) so I had to work around that, or I got a bunch of duplicated symbols errors when linking the final library together.